kbfsgit/on_demand_storer.go

```diff
@@ -23,7 +25,7 @@ var _ storage.Storer = (*onDemandStorer)(nil)
 func newOnDemandStorer(s storage.Storer) (*onDemandStorer, error) {
 	// Track a small number of recent in-memory objects, to improve
 	// performance without impacting memory too much.
-	recentCache, err := lru.New(10)
+	recentCache, err := lru.New(25)
```

Comment:
What are the usage characteristics of these objects? Is `lru` helpful? Should we consider storing a `Pool` instead?

Reply:
Also: how big are these objects usually, and how much do they vary? How big can we expect this cache to get? Should we use @songgao's memory-limited LRU queue instead of the count-limited one?

Reply:
LRU is very helpful here because of the way delta compression works.  It first sorts the objects by type and descending size, and then compares each object to a sliding window of previous objects to find a good match.  We set the size of this sliding window to 10, and so I think we were thrashing a bit in this cache because the sliding window, plus the current object, was greater than the cache size.

The sizes are the sizes of the underlying git objects themselves, which vary greatly.  However, since the set in use is based on the sliding window, it seems like that should be the limiting factor.  Eventually we might hit some repo where there's a set of large objects that can overrun memory, but at the limit that could be any two objects, and then the compression algorithm in go-git is worthless.  So I think for now I prefer this more simple thing, until we see a need for something else.

Does that sound reasonable?

Reply:
Yeah, awesome explanation @strib! Would you mind putting it into the code as a comment? IMO it'd be very helpful to have in-place if another one of us needs to pick up that part of the code.